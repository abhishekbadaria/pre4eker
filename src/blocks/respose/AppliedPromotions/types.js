const types = {
  earnedBonusPoints: "Начисление баллов",
  spentBonusPoints: "Скидка балламми",
  deliveryDiscount: "Скидка на доставку",
  message: "Сообщение",
  issuedCoupon: "Выданный промокод",
  discount: "Скидка",
};

export default types;