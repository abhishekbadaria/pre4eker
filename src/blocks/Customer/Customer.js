import React, { Component } from 'react'

import style from './Customer.css'
import Selector from '../Selector/Selector';

export default class Customer extends Component {
  constructor(props) {
    super(props)

    this.state = { 
      selected: `email`
    }

    this.options = this.props.customer
    this.changeSelection = this.changeSelection.bind(this)
  }

  changeSelection(option) {
    this.setState({
      selected: option
    })
  }

  render() {
    // console.log(this.state.selected);
    return (
      <div>
        <h2 className={style.h2}>Клиент</h2>
        <Selector
          selected={this.state.selected}
          options={this.options}
          changeSelection={this.changeSelection}
        />
      </div>
    )
  }
}
