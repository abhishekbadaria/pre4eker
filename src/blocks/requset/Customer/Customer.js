import React, { useState } from "react";
import PropTypes from "prop-types";
import style from "./Customer.css";
import Selector from "../Selector/Selector";

const Customer = ({ customer }) => {
  const [selected, changeSelection] = useState("email");

  return (
    <div>
      <h2 className={style.h2}>Клиент</h2>
      <Selector
        selected={selected}
        options={customer}
        changeSelection={(option) => changeSelection(option)}
      />
    </div>
  );
};

Customer.propTypes = {
  customer: PropTypes.object.isRequired,
};

export default Customer;
