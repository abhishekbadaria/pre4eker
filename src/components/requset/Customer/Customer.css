.inline {
  display: flex;
  margin: 0 -.3em 0 -.3em;
  flex-wrap: wrap;
  justify-content: space-between;
}

.buttonLine {
  display: flex;
  margin-bottom: .5rem;
  flex-shrink: 0;
  justify-content: space-between;
}

.half {
  width: 50%;
  padding: 1rem .3rem;
}

.third {
  width: 33%;
  padding: 0 .3rem;
}