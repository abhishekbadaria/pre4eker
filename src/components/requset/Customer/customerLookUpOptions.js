const customerLookUpOptions = [
  { type: "none", name: "Аноним" },
  { type: "ids", name: "ИД" },
  { type: "email", name: "Email" },
  { type: "mobilePhone", name: "Телефон" },
  { type: "discountCard", name: "Д. карта" },
];

export default customerLookUpOptions;
