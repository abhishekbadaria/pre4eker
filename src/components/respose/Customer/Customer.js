import React from "react";
import PropTypes from "prop-types";

import stylesShared from "src/components/respose/responseShared.css";
import Contact from "../Contact/Contact";
import statuses from "./statuses";
import sexes from "./sexes";
import CustomFields from "../CustomFields/CustomFields";

const Customer = ({ customer }) => {
  const {
    processingStatus,
    sex,
    birthDate,
    mobilePhone,
    pendingMobilePhone,
    isMobilePhoneInvalid,
    isMobilePhoneConfirmed,
    email,
    isEmailInvalid,
    isEmailConfirmed,
    pendingEmail,
    area,
    ianaTimeZone,
    timeZoneSource,
    customFields,
  } = customer;

  return (
    <div>
      <h2>Клиент</h2>
      <div className={stylesShared.customer}>
        <div className={stylesShared.line}>
          <div className={stylesShared.inline}>
            {(customer.firstName ||
              customer.middleName ||
              customer.lastName) && (
              <div className={stylesShared.half}>
                <div className={stylesShared.name}>ФИО</div>
                <div className={stylesShared.value}>
                  {customer.lastName} {customer.firstName} {customer.middleName}
                </div>
              </div>
            )}
            <div className={stylesShared.half}>
              <div className={stylesShared.name}>Статус обработки</div>
              <div className={stylesShared.value}>
                {statuses[processingStatus] || processingStatus}
              </div>
            </div>
          </div>
        </div>

        <div className={stylesShared.line}>
          <div className={stylesShared.inline}>
            {sex && (
              <div className={stylesShared.half}>
                <div className={stylesShared.name}>Пол</div>
                <div className={stylesShared.value}>{sexes[sex]}</div>
              </div>
            )}
            {birthDate && (
              <div className={stylesShared.half}>
                <div className={stylesShared.name}>Дата рождения</div>
                <div className={stylesShared.value}>{birthDate}</div>
              </div>
            )}
          </div>
        </div>

        <div className={stylesShared.line}>
          {mobilePhone && (
            <Contact
              label="Мобильный телефон"
              main={mobilePhone}
              invalid={isMobilePhoneInvalid}
              confirmed={isMobilePhoneConfirmed}
              pending={pendingMobilePhone}
            />
          )}
        </div>
        <div className={stylesShared.line}>
          {email && (
            <Contact
              label="Email"
              main={email}
              invalid={isEmailInvalid}
              confirmed={isEmailConfirmed}
              pending={pendingEmail}
            />
          )}
        </div>

        <div className={stylesShared.inline}>
          {area && (
            <div className={stylesShared.half}>
              <div className={stylesShared.name}>Зона</div>
              <div className={stylesShared.value}>{area.name}</div>
              <p className={`${stylesShared.neutral} ${stylesShared.status}`}>
                {area.ids.externalId}
              </p>
            </div>
          )}
          {ianaTimeZone && (
            <div className={stylesShared.half}>
              <div className={stylesShared.name}>Часовой пояс</div>
              <div className={stylesShared.value}>{ianaTimeZone}</div>
              <p className={`${stylesShared.neutral} ${stylesShared.status}`}>
                {timeZoneSource}
              </p>
            </div>
          )}
        </div>
        {customFields && <CustomFields data={customFields} />}
      </div>
    </div>
  );
};

Customer.propTypes = {
  customer: PropTypes.object,
};

Customer.defaultProps = {
  customer: {},
};

export default Customer;
