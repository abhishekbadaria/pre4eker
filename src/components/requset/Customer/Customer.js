import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";

import Input from "src/components/Input/Input";
import Button from "src/components/Button/Button";

import IdField from "../IdField/IdField";

import style from "./Customer.css";
import customerLookUpOptions from "./customerLookUpOptions";

const Customer = ({ setRequestJSON, requestJSON }) => {
  const [customerLookUpOption, setcustomerLookUpOption] = useState(
    customerLookUpOptions[0]
  );

  const { body } = requestJSON;

  let initialCustomer;
  if (typeof body !== "undefined") {
    initialCustomer = body.customer;
  }

  const [customer, setCustomer] = useState({ ...initialCustomer });

  useEffect(() => {
    if (Object.keys(customer).length > 0) {
      setRequestJSON({
        ...requestJSON,
        body: {
          ...body,
          customer: {
            ...customer,
          },
        },
      });
    }
  }, [customer]);

  let customerField;
  if (
    customerLookUpOption.type !== "none" &&
    customerLookUpOption.type !== "ids"
  ) {
    customerField = (
      <div className={style.inline}>
        <div className={style.half}>
          <Input
            label={customerLookUpOption.name}
            name={customerLookUpOption.type}
            value={customer[customerLookUpOption.type]}
            onChange={(e) =>
              setCustomer({
                [customerLookUpOption.type]: e.target.value,
              })
            }
          />
          <p className={style.description}>&nbsp;</p>
        </div>
      </div>
    );
  } else if (customerLookUpOption.type === "ids") {
    customerField = (
      <IdField object={customer} setObject={setCustomer} type="customer" />
    );
  }

  return (
    <>
      <div className={`${style.inline}`}>
        <h2 className={style.h2}>Клиент</h2>
        <div className={` ${style.buttonLine}`}>
          {customerLookUpOptions.map((item) => (
            <Button
              key={item.type}
              type="TEXT"
              active={customerLookUpOption.type === item.type}
              action={() => {
                setcustomerLookUpOption(item);
                setCustomer({});
                if (item.type === "none") {
                  const stateWithOutCustomer = requestJSON;
                  delete requestJSON.body.customer;
                  setRequestJSON({
                    ...stateWithOutCustomer,
                  });
                }
              }}
              size="sizeAuto"
            >
              {item.name}
            </Button>
          ))}
        </div>
      </div>
      {customerField}
    </>
  );
};

Customer.propTypes = {
  requestJSON: PropTypes.object,
  setRequestJSON: PropTypes.func,
};
Customer.defaultProps = {
  requestJSON: {},
  setRequestJSON: () => ({}),
};

export default Customer;
