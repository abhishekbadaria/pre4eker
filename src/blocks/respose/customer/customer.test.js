import React from "react";
import Customer from "./Customer";
import { render, fireEvent, waitFor, screen, cleanup, act } from '@testing-library/react' 
import response from 'Data/response'
import { exportAllDeclaration } from "@babel/types";

afterEach(cleanup)
 
const {customer} = response;

test(`customer match snapshot`, () => {
  const {asFragment, getByRole} = render(<Customer customer={customer} />)
  expect(asFragment(<Customer />)).toMatchSnapshot();

  fireEvent.click(screen.getByRole('button'));
  expect(asFragment(<Customer />)).toMatchSnapshot();
  
  fireEvent.click(screen.getByRole('button'));
  expect(asFragment(<Customer />)).toMatchSnapshot();
})