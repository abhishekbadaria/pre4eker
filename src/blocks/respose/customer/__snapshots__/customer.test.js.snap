// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer match snapshot 1`] = `
<DocumentFragment>
  <h2>
    Клиент
  </h2>
  <div
    class="customer"
  >
    <div
      class="line"
    >
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="name"
          >
            ФИО
          </div>
          <div
            class="value"
          >
            Иван Иванович Иванов
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="name"
          >
            Статус обработки
          </div>
          <div
            class="value"
          >
            Processed
          </div>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="name"
          >
            Пол
          </div>
          <div
            class="value"
          >
            Женский
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="name"
          >
            Дата рождения
          </div>
          <div
            class="value"
          >
            1990-01-31
          </div>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="name"
      >
        Мобильный телефон
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="value"
          >
            79001234567
          </div>
          <p
            class="good status"
          >
            Валидный 
          </p>
          <p
            class="good status"
          >
            Подтвержден 
          </p>
        </div>
        <div
          class="half"
        >
          <div
            class="value"
          >
            79652026149
          </div>
          <p
            class="bad status"
          >
            Ожидает подтверждения 
          </p>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="name"
      >
        Email
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="value"
          >
            test@mindbox.ru
          </div>
          <p
            class="good status"
          >
            Валидный 
          </p>
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Зона
        </div>
        <div
          class="value"
        >
          Москва
        </div>
        <p
          class="neutral status"
        >
          MoscowCity
        </p>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Часовой пояс
        </div>
        <div
          class="value"
        >
          Asia/Hong_Kong
        </div>
        <p
          class="neutral status"
        >
          Определили в трекере
        </p>
      </div>
    </div>
    <div>
      <div
        class="name"
      >
        Дополнительные поля
        <button
          class="small button normal undefined basic"
        >
          <div>
            <svg
              fill="none"
              height="5"
              viewBox="0 0 8 5"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`customer match snapshot 2`] = `
<DocumentFragment>
  <h2>
    Клиент
  </h2>
  <div
    class="customer"
  >
    <div
      class="line"
    >
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="name"
          >
            ФИО
          </div>
          <div
            class="value"
          >
            Иван Иванович Иванов
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="name"
          >
            Статус обработки
          </div>
          <div
            class="value"
          >
            Processed
          </div>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="name"
          >
            Пол
          </div>
          <div
            class="value"
          >
            Женский
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="name"
          >
            Дата рождения
          </div>
          <div
            class="value"
          >
            1990-01-31
          </div>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="name"
      >
        Мобильный телефон
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="value"
          >
            79001234567
          </div>
          <p
            class="good status"
          >
            Валидный 
          </p>
          <p
            class="good status"
          >
            Подтвержден 
          </p>
        </div>
        <div
          class="half"
        >
          <div
            class="value"
          >
            79652026149
          </div>
          <p
            class="bad status"
          >
            Ожидает подтверждения 
          </p>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="name"
      >
        Email
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="value"
          >
            test@mindbox.ru
          </div>
          <p
            class="good status"
          >
            Валидный 
          </p>
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Зона
        </div>
        <div
          class="value"
        >
          Москва
        </div>
        <p
          class="neutral status"
        >
          MoscowCity
        </p>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Часовой пояс
        </div>
        <div
          class="value"
        >
          Asia/Hong_Kong
        </div>
        <p
          class="neutral status"
        >
          Определили в трекере
        </p>
      </div>
    </div>
    <div>
      <div
        class="name"
      >
        Дополнительные поля
        <button
          class="small button normal undefined basic"
        >
          <div>
            <svg
              fill="none"
              height="5"
              viewBox="0 0 8 5"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          child1Sex
        </div>
        <div
          class="half value"
        >
          <span
            class="neutral"
          >
            Пусто
          </span>
        </div>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          clearTestClient
        </div>
        <div
          class="half value"
        >
          <span
            class="neutral"
          >
            Пусто
          </span>
        </div>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          isCompany
        </div>
        <div
          class="half value"
        >
          true
        </div>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          multiline
        </div>
        <div
          class="half value"
        >
          <div>
            1
          </div>
          <div>
            2
          </div>
        </div>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          singleline
        </div>
        <div
          class="half value"
        >
          <div>
            56
          </div>
          <div>
            <span
              class="neutral"
            >
              Пусто
            </span>
          </div>
        </div>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          walletCardLink
        </div>
        <div
          class="half value"
        >
          <span
            class="neutral"
          >
            Пусто
          </span>
        </div>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          walletCardQrCodeLink
        </div>
        <div
          class="half value"
        >
          <span
            class="neutral"
          >
            Пусто
          </span>
        </div>
      </div>
      <div
        class="inline line"
      >
        <div
          class="half value"
        >
          walletCardSubscription
        </div>
        <div
          class="half value"
        >
          false
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`customer match snapshot 3`] = `
<DocumentFragment>
  <h2>
    Клиент
  </h2>
  <div
    class="customer"
  >
    <div
      class="line"
    >
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="name"
          >
            ФИО
          </div>
          <div
            class="value"
          >
            Иван Иванович Иванов
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="name"
          >
            Статус обработки
          </div>
          <div
            class="value"
          >
            Processed
          </div>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="name"
          >
            Пол
          </div>
          <div
            class="value"
          >
            Женский
          </div>
        </div>
        <div
          class="half"
        >
          <div
            class="name"
          >
            Дата рождения
          </div>
          <div
            class="value"
          >
            1990-01-31
          </div>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="name"
      >
        Мобильный телефон
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="value"
          >
            79001234567
          </div>
          <p
            class="good status"
          >
            Валидный 
          </p>
          <p
            class="good status"
          >
            Подтвержден 
          </p>
        </div>
        <div
          class="half"
        >
          <div
            class="value"
          >
            79652026149
          </div>
          <p
            class="bad status"
          >
            Ожидает подтверждения 
          </p>
        </div>
      </div>
    </div>
    <div
      class="line"
    >
      <div
        class="name"
      >
        Email
      </div>
      <div
        class="inline"
      >
        <div
          class="half"
        >
          <div
            class="value"
          >
            test@mindbox.ru
          </div>
          <p
            class="good status"
          >
            Валидный 
          </p>
        </div>
      </div>
    </div>
    <div
      class="inline"
    >
      <div
        class="half"
      >
        <div
          class="name"
        >
          Зона
        </div>
        <div
          class="value"
        >
          Москва
        </div>
        <p
          class="neutral status"
        >
          MoscowCity
        </p>
      </div>
      <div
        class="half"
      >
        <div
          class="name"
        >
          Часовой пояс
        </div>
        <div
          class="value"
        >
          Asia/Hong_Kong
        </div>
        <p
          class="neutral status"
        >
          Определили в трекере
        </p>
      </div>
    </div>
    <div>
      <div
        class="name"
      >
        Дополнительные поля
        <button
          class="small button normal undefined basic"
        >
          <div>
            <svg
              fill="none"
              height="5"
              viewBox="0 0 8 5"
              width="8"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.0344238 1.16905L0.732916 0.502381L3.54531 3.1561L6.33428 0.499939L7.03473 1.16647L3.53314 4.50002L0.0344238 1.16905Z"
                fill="#000"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
